File: BackEnd/src/domain/interfaces/usecases/community/IGetPostByIdUseCase.ts
/**
 * @file IGetPostByIdUseCase.ts
 * @description Interfaz del caso de uso para obtener un post por ID.
 * @module Community
 * @layer Domain
 */

import type { PostResponseDTO } from '../../../dtos/community/post-response.dto.js';

/**
 * Contrato del use case GetPostById.
 * Devuelve un post enriquecido con datos del autor.
 */
export interface IGetPostByIdUseCase {
  /**
   * Ejecuta el caso de uso.
   *
   * @param postId           — Id del post a obtener.
   * @param requestingUserId — Id del usuario que solicita (para calcular isLikedByMe).
   * @returns PostResponseDTO con isLikedByMe correcto.
   * @throws {NotFoundException} Si el post no existe o está eliminado.
   */
  execute(postId: string, requestingUserId?: string): Promise<PostResponseDTO>;
}

File: FrontEnd/lib/domain/interfaces/usecases/community/i_get_post_by_id_use_case.dart
/// @file i_get_post_by_id_use_case.dart
/// @description Interfaz: Obtiene un post por su ID.
/// @module Community
/// @layer Domain
library;

import '../../../entities/post.dart';

abstract interface class IGetPostByIdUseCase {
  /// Obtiene el detalle de un post por su identificador.
  ///
  /// [param] postId — Identificador del post.
  /// [returns] [Post] completo con todos sus campos.
  Future<Post> execute(String postId);
}
